core/src/avm2/globals/flash/net.rs

```diff
@@ -15,6 +17,47 @@ pub mod socket;
 pub mod url_loader;
 pub mod xml_socket;
 
+fn object_to_index_map<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    obj: &Object<'gc>,
+) -> Result<IndexMap<String, String>, Error<'gc>> {
+    let mut map = IndexMap::new();
+    let mut last_index = obj.get_next_enumerant(0, activation)?;
+    while let Some(index) = last_index {
+        let name = obj
+            .get_enumerant_name(index, activation)?
+            .coerce_to_string(activation)?;
+        let value = obj
+            .get_public_property(name, activation)?
+            .coerce_to_string(activation)?
+            .to_utf8_lossy()
+            .to_string();
+
+        let name = name.to_utf8_lossy().to_string();
+        map.insert(name, value);
+        last_index = obj.get_next_enumerant(index, activation)?;
+    }
+    Ok(map)
+}
+
+fn parse_data<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    url: &String,
+    data: &Value<'gc>,
+) -> Result<(String, IndexMap<String, String>), Error<'gc>> {
+    let mut url = url.to_string();
+    let obj = data.coerce_to_object(activation)?;
+    let vars = object_to_index_map(activation, &obj).unwrap_or_default();
```

Comment:
Okay one last thing. I am pretty sure we are only supposed to handle URLVariables as a map and coerce everything else to a string.

```rust
let urlvariables = activation
                                .avm2()
                                .classes()
                                .urlvariables
if data.is_type_of(urlvariables.inner_class_definition(), &mut activation.context) {
...
  object_to_index_map()
...
} else {
 ...
  coerce_to_string()
  ...
}
```

Reply:
good call! Was not aware you could check type explicitly like that. Changed in https://github.com/ruffle-rs/ruffle/pull/15129/commits/9472c52dcbb84b0911b2bf31105fe70b71e9c501